etcdserver/server.go

```diff
@@ -527,6 +539,62 @@ func NewServer(cfg ServerConfig) (srv *EtcdServer, err error) {
 	}
 	srv.r.transport = tr
 
+	// fresh start
+	if !haveWAL {
+		ticks := cfg.ElectionTicks - 1
+		plog.Infof("%s started anew; fast-forwarding %d ticks (election ticks %d) with %d found member(s)", srv.ID(), ticks, cfg.ElectionTicks, len(cl.Members()))
+		srv.advanceRaftTicks(ticks)
+		return srv, nil
+	}
+
+	srv.goAttach(func() {
+		select {
+		case <-cl.InitialAddNotify():
```

Comment:
this is pretty complicated. let us just get the peer list from the existing snapshot. we do not need to ensure all the configuration in the wal file are executed.

Reply:
I was trying to cover all cases where there's no snapshot (which needs to populate member lists from WAL). But, agree that this should be simplified by loading members from snapshot. Will rework on this.